contrib/golang/filters/http/source/go/pkg/http/shim.go

```diff
@@ -208,6 +209,9 @@ func envoyGoFilterOnHttpDestroy(r *C.httpRequest, reason uint64) {
 	req := getRequest(r)
 	// do nothing even when req.panic is true, since filter is already destroying.
 	defer req.RecoverPanic()
+	if atomic.LoadInt32(&req.waitingOnEnvoy) != 0 {
```

Comment:
there might be an extremely little race.
We can not assume `waitingOnEnvoy` will `-` in:
https://github.com/envoyproxy/envoy/pull/26883/files#diff-d0977c943942fbbf3c44b7e48ce284c00116a0b6b66cc3c3a26687959d31d84aR234
after we invoke `sema.Done` in:
https://github.com/envoyproxy/envoy/pull/26883/files#diff-7216c7bf23980611fbbeebf5559ef1835bbbec0db4e5e2a146b671d15b73aeecR234

`sema.Done` just make the waiting goroutine is ready, it may be scheduled after a piece of time.
after envoy worker thread invoked `onDestroy`, an accident happens.

invoke `sema.Done` twice will cause a panic, when only `Add(1)` once. the following `OnDestroy` and `DeleteReq` will be skipped.